tests/dummy_qrc/qblox.py

```diff
@@ -38,189 +24,108 @@
-    qcm_rf1 = instantiate_module(
-        modules, ClusterQCM_RF, "qcm_rf1", "192.168.0.6:3", instruments_settings
-    )  # qubits q1, q2
-    qcm_rf2 = instantiate_module(
-        modules, ClusterQCM_RF, "qcm_rf2", "192.168.0.6:4", instruments_settings
-    )  # qubits q3, q4
-
-    qcm_bb0 = instantiate_module(modules, ClusterQCM_BB, "qcm_bb0", "192.168.0.6:5", instruments_settings)  # qubit q0
-    qcm_bb1 = instantiate_module(
-        modules, ClusterQCM_BB, "qcm_bb1", "192.168.0.6:2", instruments_settings
-    )  # qubits q1, q2, q3, q4
-
     # DEBUG: debug folder = report folder
     # import os
     # folder = os.path.dirname(runcard) + "/debug/"
     # if not os.path.exists(folder):
     #     os.makedirs(folder)
     # for name in modules:
     #     modules[name]._debug_folder = folder
+    qcm_bb0 = ClusterQCM_BB("qcm_bb0", f"{ADDRESS}:2", cluster)
+    qcm_bb1 = ClusterQCM_BB("qcm_bb1", f"{ADDRESS}:4", cluster)
+    qcm_rf0 = ClusterQCM_RF("qcm_rf0", f"{ADDRESS}:6", cluster)
+    qcm_rf1 = ClusterQCM_RF("qcm_rf1", f"{ADDRESS}:8", cluster)
+    qcm_rf2 = ClusterQCM_RF("qcm_rf2", f"{ADDRESS}:10", cluster)
+    qrm_rf_a = ClusterQRM_RF("qrm_rf_a", f"{ADDRESS}:16", cluster)
+    qrm_rf_b = ClusterQRM_RF("qrm_rf_b", f"{ADDRESS}:18", cluster)
 
     controller = QbloxController("qblox_controller", cluster, modules)
 
-    twpa_pump = SGS100A(name="twpa_pump", address="192.168.0.37")
-    twpa_pump.frequency = instruments_settings["twpa_pump"]["frequency"]
-    twpa_pump.power = instruments_settings["twpa_pump"]["power"]
+    twpa_pump = SGS100A(name="twpa_pump", address="192.168.0.36")
+    instruments = {
+        controller.name: controller,
+        twpa_pump.name: twpa_pump,
+        qcm_bb0.name: qcm_bb0,
+        qcm_bb1.name: qcm_bb1,
+        qcm_rf0.name: qcm_rf0,
+        qcm_rf1.name: qcm_rf1,
+        qcm_rf2.name: qcm_rf2,
+        qrm_rf_a.name: qrm_rf_a,
+        qrm_rf_b.name: qrm_rf_b,
+    }
+
+    instruments = load_instrument_settings(runcard, instruments)
+
+    modules["qcm_bb0"] = qcm_bb0
+    modules["qcm_bb1"] = qcm_bb1
+    modules["qcm_rf0"] = qcm_rf0
+    modules["qcm_rf1"] = qcm_rf1
+    modules["qcm_rf2"] = qcm_rf2
+    modules["qrm_rf_a"] = qrm_rf_a
+    modules["qrm_rf_b"] = qrm_rf_b
 
     # Create channel objects
     channels = {}
-    # readout
-    channels["L3-25_a"] = Channel(name="L3-25_a", port=qrm_rf_a.ports["o1"])
-    channels["L3-25_b"] = Channel(name="L3-25_b", port=qrm_rf_b.ports["o1"])
 
-    # feedback
-    channels["L2-5_a"] = Channel(name="L2-5_a", port=qrm_rf_a.ports["i1"])
-    channels["L2-5_b"] = Channel(name="L2-5_b", port=qrm_rf_b.ports["i1"])
+    def instantiate_channels(channel_name: str, module, module_port_name: str):
+        module.channels.append(channel_name)
+        module._port_channel_map[module_port_name] = channel_name
+        module._channel_port_map[channel_name] = module_port_name
+        return Channel(name=channel_name, port=module.ports[module_port_name])
```

Comment:
The main reason I want to drop `_port_channel_map` and `_channel_port_map` is to get rid of this function. From the user perspective (the one that is writing the `create` method) it should be sufficient to define the channels and map them to ports, ie
```py
channels["L3-15"] = Channel(name="L3-15", port=qcm_rf0.ports["o1"])
```
This contains all the information we need from the lab configuration. The rest are manipulations of the channel map that we can do internally and only when needed.